File: C#/RefactoringWorkshop/RefactoringWorkshop/Coupler/MessageChain/ArticleWrapperConverter.cs
namespace RefactoringWorkshop.Coupler.MessageChain
{
    public class ArticleWrapperConverter
    {
        public ArticleWrapper Convert(ArticleResponseWrapper articleResponseWrapper) {
            var articleWrapper = new ArticleWrapper();

            if (articleResponseWrapper.ArticleResponse.Articles.Count > 0) {
                var article = articleResponseWrapper.ArticleResponse.Articles[0];
                articleWrapper.ArticleRef = article.ArticleRef;

                if (articleResponseWrapper.CompanySubscription != null) {
                    foreach (var subscriptionPreferenceWrapper in articleResponseWrapper.CompanySubscription.SubscriptionPreferences) {
                        if (article.MetadataCodes != null) {
                            if (article.MetadataCodes.GetCompanyCodes().ContainsKey(subscriptionPreferenceWrapper.EntityCode.ToUpper())
                                || article.MetadataCodes.GetCompanyCodes()
                                    .ContainsKey(subscriptionPreferenceWrapper.EntityCode.ToLower())) {
                                articleWrapper.Companies.Add(subscriptionPreferenceWrapper.EntityName);
                                articleWrapper.EntityCodes.Add(subscriptionPreferenceWrapper.EntityCode.ToUpper());
                            }
                        }
                    }
                }
            }
            return articleWrapper;
        }

    }
}

File: Java/RefactoringWorkshop/src/main/java/coupler/messagechain/ArticleWrapperConverter.java
package coupler.messagechain;

public class ArticleWrapperConverter {

    public ArticleWrapper convert(ArticleResponseWrapper articleResponseWrapper) {
        ArticleWrapper articleWrapper = new ArticleWrapper();

        if (!articleResponseWrapper.getArticleResponse().getArticles().isEmpty()) {
            Article article = articleResponseWrapper.getArticleResponse().getArticles().get(0);
            articleWrapper.setArticleRef(article.getArticleRef());

            if (articleResponseWrapper.getCompanySubscription() != null) {
                for (SubscriptionPreferenceWrapper subscriptionPreferenceWrapper : articleResponseWrapper.getCompanySubscription()
                        .getSubscriptionPreferences()) {
                    if (article.getMetadataCodes() != null) {
                        if (article.getMetadataCodes().getCompanyCodes().containsKey(subscriptionPreferenceWrapper.getEntityCode().toUpperCase())
                                || article.getMetadataCodes().getCompanyCodes()
                                .containsKey(subscriptionPreferenceWrapper.getEntityCode().toLowerCase())) {
                            articleWrapper.addCompanies(subscriptionPreferenceWrapper.getEntityName());
                            articleWrapper.addEntityCodes(subscriptionPreferenceWrapper.getEntityCode().toUpperCase());
                        }
                    }
                }
            }
        }
        return articleWrapper;
    }
}

